arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/ParZipOrAccumulate.kt

```diff
@@ -0,0 +1,48 @@
+package arrow.fx.coroutines
+
+import arrow.core.Either
+import arrow.core.NonEmptyList
+import arrow.core.raise.either
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.coroutineScope
+
+public suspend inline fun <E, A, B, C> parZipOrAccumulate(
+  combine: (E, E) -> E,
+  crossinline fa: suspend ScopedRaise<E>.() -> A,
+  crossinline fb: suspend ScopedRaise<E>.() -> B,
+  crossinline f: suspend CoroutineScope.(A, B) -> C
+): Either<E, C> =
+  parZipOrAccumulate(EmptyCoroutineContext, combine, fa, fb, f)
+
+public suspend inline fun <E, A, B, C> parZipOrAccumulate(
+  ctx: CoroutineContext,
+  combine: (E, E) -> E,
+  crossinline fa: suspend ScopedRaise<E>.() -> A,
+  crossinline fb: suspend ScopedRaise<E>.() -> B,
```

Comment:
This can still benefit from the `ScopedAccumulatingRaise` type from https://github.com/arrow-kt/arrow/pull/2952, and makes me wonder if `Either.zipOrAccumulate` shouldn't be DSL based as well to support both `E` and `Nel<E>` in a single API. 🤔

Currently we support both `Either` and `EitherNel` separately, and they both types don't interop with each-other.